echo-telemetry/src/test/groovy/com/netflix/spinnaker/echo/telemetry/TelemetryEventListenerSpec.groovy

```diff
@@ -367,6 +367,77 @@ class TelemetryEventListenerSpec extends Specification {
     }
   }
 
+  def "test event has deploy method information happy path"(String deploymentType, String deployVersion) {
+      given:
+      def configProps = new TelemetryConfig.TelemetryConfigProps()
+        .setInstanceId(instanceId)
+        .setSpinnakerVersion(spinnakerVersion)
+        .setDeploymentMethod([
+          "type": deploymentType,
+          "version": deployVersion
+        ])
+      def deployEnumType = DeploymentMethod.Type.valueOf(deploymentType.toUpperCase())
+
+      @Subject
+      def listener = new TelemetryEventListener(service, configProps, registry)
+
+      when:
+      listener.processEvent(validEvent)
+
+      then:
+      1 * service.log(_) >> { List args ->
+        String body = args[0]?.toString()
+        assert body != null
+
+        // Note the handy Groovy feature of import aliasing Event->EventProto
+        EventProto.Builder eventBuilder = EventProto.newBuilder()
+        JsonFormat.parser().merge(body, eventBuilder)
+
+        EventProto e = eventBuilder.build()
+        assert e != null
+
+        SpinnakerInstance s = e.spinnakerInstance
+        assert s != null
+        assert s.deploymentMethod != null
+        assert s.deploymentMethod.type == deployEnumType
+        assert s.deploymentMethod.version == deployVersion
+      }
+
+      where:
```

Comment:
I'd make this a test table if you're going to use that many pairs. 

http://spockframework.org/spock/docs/1.0/data_driven_testing.html#data-tables

Reply:
Are there examples in the Spinnaker project of property/generative testing? Ideally we'd have tests for:
- expected enum values
- random strings that get passed in for type (and set to the appropriate default value)
- validating arbitrary version strings don't break the config object.

Reply:
I don't think so. It would be great if we could "fuzz test" our APIs, but AFAIK we don't have that.